test/test_data/test_validation.py

```diff
@@ -229,3 +294,234 @@ def test_dataframe_econding_2D(self):
 
         y_decoded = validator.decode_target(y)
         np.testing.assert_array_almost_equal(y, y_decoded)
+
+    def test_noNaN(self):
+        """
+        Makes sure that during classification/regression task,
+        the transformed data is not corrupted.
+
+        Testing is given without Nan and no sparse data
+        """
+        # numpy - categorical - classification
+        x = np.array(['a', 'b', 'c', 'a', 'b', 'c']).reshape(-1, 1)
```

Comment:
Sorry, I'm just seeing why you pass the data like this - by my last comment I meant with 'numpy, categorical' that we have an array that has integer-encoded categories inside a numerical array. Thinking about this again, if it is a numpy array, we do not need to check it at all as the categories are not encoded, right?

Maybe we should check that the numbers are actually integers for columns which are categories.

Reply:
I think we should not do anything right? Because the pipeline should be able to handle numerical data at the input, be it integer or float?

Maybe the reason I am saying this is that to know whether a feature is categorical or not, we will rely on the user input (feature_types) in case we don't have a pandas frame. And even without this check, the older autosklearn code was able to handle this?

Reply:
> I think we should not do anything right?

Reply:
True, the requirement for the encoding was an old version of the OneHotEncoder which was only able to handle positive integers

Reply:
Okay, and even that statement is actually wrong given that we encode the data prior to passing it to the pipeline.